packages/esm-patient-labs-app/src/lab-orders/add-lab-order/test-type-search.component.tsx

```diff
@@ -84,61 +104,60 @@ function TestTypeSearchResults({ searchTerm, openOrderForm, focusAndClearSearchI
     );
   }
 
-  return (
-    <>
-      {testTypes?.length ? (
+  if (filteredTestTypes?.length) {
+    return (
+      <>
         <div className={styles.container}>
           {searchTerm && (
             <div className={styles.orderBasketSearchResultsHeader}>
               <span className={styles.searchResultsCount}>
                 {t('searchResultsMatchesForTerm', '{{count}} results for "{{searchTerm}}"', {
-                  count: testTypes?.length,
+                  count: filteredTestTypes?.length,
                   searchTerm,
                 })}
               </span>
               <Button kind="ghost" onClick={focusAndClearSearchInput} size={isTablet ? 'md' : 'sm'}>
-                {t('clearSearchResults', 'Clear Results')}
+                {t('clearSearchResults', 'Clear results')}
               </Button>
             </div>
           )}
           <div className={styles.resultsContainer}>
-            {testTypes.map((testType) => (
-              <TestTypeSearchResultItem key={testType.conceptUuid} testType={testType} openOrderForm={openOrderForm} />
+            {filteredTestTypes.map((testType, index) => (
+              <TestTypeSearchResultItem key={index} openOrderForm={openOrderForm} t={t} testType={testType} />
```

Comment:
Ideally the key should come from the [data from the backend](https://react.dev/learn/rendering-lists#where-to-get-your-key), but we need to fix the issue with duplicates first.